### DISCUSSION TITLE
index is not uniformly distributed to every k8s indexnode

### DISCUSSION THREAD

**Discussion by xiaojuan8** [milvus-io/milvus#16075] [Q&A and General discussion]
Hi，I have 100 milllion vectors (dimension=32), I inserted these data to milvus with 5 nodes, but while I was building index, index node finished their indexing task with different time cost, I check k8s logs and found that these indexnode workers got different amount of indexing task, and the last node finished its' indexing much slower than the other ones, I expect every node finishes indexing probably at the same time. k8s logs shown as below:

<img width="1240" alt="image" src="https://user-images.githubusercontent.com/74544828/158565475-a1e03675-cee7-4748-aaa8-529800ea2196.png">

**Comment by xiaofan-luan** [comment]
Hi @xiaojuan8 , if all indexnode is free, milvus randomly assign task to one of the indexnode, each indexnode will only take one of the task in one run.

Look at the your log, my guess is you don't need much 5 indexnodes and most of the index are free, So the system randomly assign task to one of the index node, I'm not sure why one of the node has such priority, but in your case 2-3 indexnodes should be well enough

**Reply by xiaojuan8** [reply]
@xiaofan-luan @yhmo Is it possible a bug of task schedule? Because the last node is indexing all the way since 18 hours have passed after the other indexnodes finished their indexing.  The CPU monitor is shown below :
![image](https://user-images.githubusercontent.com/74544828/158727686-c9be16c1-52c8-4b8d-abff-d537aa8ca814.png)
<img width="1240" alt="image" src="https://user-images.githubusercontent.com/74544828/158728444-e6d18fd1-28f0-4114-8189-e0cfbf76df87.png">
